js/app/modules/selection/subsetsForSet.js

```diff
@@ -10,13 +12,27 @@ const Xapian = imports.app.modules.selection.xapian;
 const SubsetsForSet = new Module.Class({
     Name: 'Selection.SubsetsForSet',
     Extends: Xapian.Xapian,
+    Properties: {
+        /**
+         * Property: top-level-only
+         * Whether this content group has more content to show
+         */
+        'top-level-only': GObject.ParamSpec.boolean('top-level-only',
+            'Top level only', 'Only refresh when top level set changes',
+            GObject.ParamFlags.READWRITE | GObject.ParamFlags.CONSTRUCT_ONLY,
+            false),
+    },
 
     _init: function (props={}) {
         this.parent(props);
         if (this.global) {
-            this.model = HistoryStore.get_default().current_set;
-            HistoryStore.get_default().connect('notify::current-set', () => {
-                this.model = HistoryStore.get_default().current_set;
+            let property_name = 'current-set';
+            if (this.top_level_only) {
+                property_name = 'current-top-level-set';
+            }
+            this.model = HistoryStore.get_default()[property_name.replace('-', '_', 'g')];
```

Comment:
Did you ever try to that gjs patch? Would love to fix this up without carry workarounds for gjs quirks...